### DISCUSSION TITLE
Attach VS Code via CLI

### DISCUSSION THREAD

**Discussion by devhl-labs** [microsoft/vscode-docker#3710] [Q&A] [answered]
I start my work day by opening a Windows Terminal profile. This profile has a command that opens code to my working directory. I then manually have to go to the docker extension, right click on my container, and click attach VS Code. Can I tell the extension to attach to the container via the CLI to avoid doing it manually?

**Comment by karolz-ms** [comment, marked as answer]
@devhl-labs have you considered using a [dev container](https://code.visualstudio.com/docs/devcontainers/containers)? Once you configure a dev container for particular folder/workspace, VS Code will automatically open that folder in container as a default.

Also FYI, the "attach VS Code" function is provided by Dev Containers extension, not Docker extension, so if you encounter any problems with using it (or any problems with dev containers), the best repo to open an issue/ask questions is https://github.com/microsoft/vscode-remote-release

Let us know please if that helps

**Reply by bwateratmsft** [reply]
I don't know for sure but I *think* I recall the remote extensions having CLI auto-attach capability, but https://github.com/microsoft/vscode-remote-release would be the place to ask.

**Reply by devhl-labs** [reply]
Thanks for the replies. It looks like I need this feature added to attach to an already running container. https://github.com/devcontainers/spec/issues/20

Here is another issue that talks about it over there.
https://github.com/microsoft/vscode-remote-release/issues/5278
